docs/development/defaulting.md

```diff
@@ -0,0 +1,24 @@
+# Defaulting Strategy and Developer Guideline
+
+This document walks you through:
+
+- Conventioned to be followed when writing defaulting functions
+- How to write test for defaulting function
+
+The document is aimed towards developers that want to contribute code and need to write defaulting code and unit tests covering the defaulting functions, as well as maintainers and reviewers that review code.
+It serves as a common guide that we commit to follow in our project to ensure consistency in our defaulting code, good coverage for high confidence, and good maintainability.
+
+## Writing defaulting code
+
+- Every Kubernetes type should have a dedicated `defaults_*.go` file. For instance, if you have a `Shoot` type, there should be a corresponding `defaults_shoot.go` file containing all defaulting logic for that type.
+- If there is only one type under an api group then we can just have `types.go` and a corresponding `defaults.go` file.
+- Aim to segregate each struct type into its own `SetDefaults_*` function. These functions encapsulate the defaulting logic specific to the corresponding struct type, enhancing modularity and maintainability.
+
+:warning: Ensure to run the `make generate` command, which generates the `zz_generated.defaults.go` file containing the overall defaulting function.
```

Comment:
I guess you rather mean that `make generate` has to be executed after a new `SetDefaults_*` func is added? Otherwise, reading this sentence I don't understand when I have to execute `make generate`.

Reply:
Additionally, there should be a more effective way of generation, for example executing `make generate WHAT=codegen` instead of `make generate`.